src/utils/bit-utils.js

```diff
@@ -0,0 +1,98 @@
+import { defineQuery, hasComponent } from "bitecs";
+import { $isStringType, NetworkedMediaFrame } from "../bit-components";
+
+const queries = new Map();
+export function anyEntityWith(world, component) {
+  if (!queries.has(component)) {
+    queries.set(component, defineQuery([component]));
+  }
+
+  const eids = queries.get(component)(world);
+  return eids.length && eids[0];
+}
+
+export function hasAnyComponent(world, components, eid) {
+  for (let i = 0; i < components.length; i++) {
+    if (hasComponent(world, components[i], eid)) return true;
+  }
+  return false;
+}
+
+// TODO HACK gettting internal bitecs symbol, should expose createShadow
+const $parentArray = Object.getOwnPropertySymbols(NetworkedMediaFrame.scale).find(s => s.description == "parentArray");
+const $storeFlattened = Object.getOwnPropertySymbols(NetworkedMediaFrame).find(s => s.description == "storeFlattened");
+
+const createShadow = (store, key) => {
+  if (!ArrayBuffer.isView(store)) {
+    const shadowStore = store[$parentArray].slice(0);
+    store[key] = store.map((_, eid) => {
+      const { length } = store[eid];
+      const start = length * eid;
+      const end = start + length;
+      return shadowStore.subarray(start, end);
+    });
+  } else {
+    store[key] = store.slice(0);
+  }
+  return key;
+};
+
+// TODO this array encoding is silly, use a buffer once we are not sending JSON
```

Comment:
This is about as noisy as our current JSON encoding in NAF... It should be quite easy to make this much better when we move to a binary socket.